test/unit/query_thread.c

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2018 Intel Corporation. All rights reserved.
+ *  This software is available to you under the BSD license below:
+ *
+ *      Redistribution and use in source and binary forms, with or
+ *      without modification, are permitted provided that the following
+ *      conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <assert.h>
+#include <shmem.h>
+
+int
+main(int argc, char* argv[])
+{
+    int provided;
+
+#if defined(ENABLE_THREADS)
```

Comment:
oh. right. missed that.

Reply:
@jdinan, correct me if I am wrong, with ```--disable-threads``` the ```query_thread_funneled``` test would always fail unless added with an ```if ENABLE_THREADS``` check in Makefile, right?

Reply:
Sorry, I missed this question -- yes that is correct.

Reply:
Could you create another PR to fix this -- good catch!

Reply:
Sure. Will do.

Reply:
Sorry, was thinking this was merged already -- just push the fix here.  :)

Reply:
@jdinan, one more question. Currently, ```ENABLE_THREADS``` is not added in AM_CONDITIONAL, but ```HAVE_PTHREADS``` is. Although this program does not really require pthreads, we can add the check with pthreads for this one as well just to avoid updating the configure.ac. Let me know your thoughts.

Reply:
I'm fine with that.  The build system is complicated enough as it is.  :)